### PR TITLE
Enhance errors

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: istio/error.go [hunk 1]
```diff
@@ -133,11 +133,11 @@ var (
...
 
 	// ErrParseOAMConfig represents the error which is
 	// generated during the OAM configuration parsing
-	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while prasing component config in the OAM request made"}, []string{"Invalid OAM config passed in OAM request"}, []string{"Check if your request has vaild OAM config"})
+	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while parsing application configuration in the OAM request made by meshery server"}, []string{"Could not unmarshall OAM config recieved via ProcessOAM gRPC call into a valid Config struct"}, []string{"Check if meshery server is creating valid config for ProcessOAM gRPC call. This error should never happen and can be reported as a bug in meshery server. Also check if meshery server and adapters are referring to same config struct provided in meshkit"})
```
Comment (leecalcote) [inline istio/error.go:140]:
```suggestion
	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while parsing application configuration in the OAM request made by Meshery Server"}, []string{"Could not unmarshall OAM config recieved via ProcessOAM gRPC call into a valid Config struct"}, []string{"Check if Meshery Server is creating valid config for ProcessOAM gRPC call. This error should never happen and can be reported as a bug in Meshery Server. Also, confirm that Meshery Server and Adapters are referring to same config struct provided in MeshKit"})
```

**Review by leecalcote — COMMENTED** [review]
Thanks. Some capitalization needed in both messages.

**Comment by Revolyssup** [comment]
@asubedy  The error message that I changed here exists in all other adapters. WIll you mind copy pasting this change across each of the adapters which  have this error message

**Comment by asubedy** [comment]
Ok on it

File: istio/error.go [hunk 2]
```diff
@@ -133,11 +133,11 @@ var (
 
 	// ErrParseOAMComponent represents the error which is
 	// generated during the OAM component parsing
-	ErrParseOAMComponent = errors.New(ErrParseOAMComponentCode, errors.Alert, []string{"error parsing the component"}, []string{"Error occured while prasing application component in the OAM request made"}, []string{"Invalid OAM component passed in OAM request"}, []string{"Check if your request has vaild OAM components"})
+	ErrParseOAMComponent = errors.New(ErrParseOAMComponentCode, errors.Alert, []string{"error parsing the component"}, []string{"Error occured while parsing application component in the OAM request made by meshery server"}, []string{"Could not unmarshall OAM component recieved via ProcessOAM gRPC call into a valid Component struct"}, []string{"Check if meshery server is creating valid component for ProcessOAM gRPC call. This error should never happen and can be reported as a bug in meshery server. Also check if meshery server and adapters are referring to same component struct provided in meshkit."})
```
Comment (leecalcote) [inline istio/error.go:136]: Capitalize Meshery Server

File: istio/error.go [hunk 3]
```diff
@@ -133,11 +133,11 @@ var (
...
 
 	// ErrParseOAMConfig represents the error which is
 	// generated during the OAM configuration parsing
-	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while prasing component config in the OAM request made"}, []string{"Invalid OAM config passed in OAM request"}, []string{"Check if your request has vaild OAM config"})
+	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while parsing application configuration in the OAM request made by Meshery Server"}, []string{"Could not unmarshall OAM config recieved via ProcessOAM gRPC call into a valid Config struct"}, []string{"Check if Meshery Server is creating valid config for ProcessOAM gRPC call. This error should never happen and can be reported as a bug in Meshery Server. Also, confirm that Meshery Server and Adapters are referring to same config struct provided in MeshKit"})
```
Comment (leecalcote) [inline istio/error.go:140]:
```suggestion
	ErrParseOAMConfig = errors.New(ErrParseOAMConfigCode, errors.Alert, []string{"error parsing the configuration"}, []string{"Error occured while parsing configuration in the request made by Meshery Server"}, []string{"Could not unmarshall OAM config recieved via ProcessOAM gRPC call into a valid Config struct"}, []string{"Check if Meshery Server is creating valid config for ProcessOAM gRPC call. This error should never happen and can be reported as a bug in Meshery Server. Also, confirm that Meshery Server and Adapters are referring to same config struct provided in MeshKit"})
```

Note:

- the request is most likely made by Meshery Server, but could be another gRPC client. Meshery Server is the intended and supported client. 
- we can downplay OAM somewhat.

**Comment by Revolyssup** [comment]
@leecalcote  Can you approve this?

**Review by leecalcote — APPROVED** [review]
LGTM

**Pull request merged**